### PR TITLE
Generate new operators in MongoDB 3.0

### LINKED ISSUE #1310: MongoDB 3.0: use new operators
Depends on #1308

There is only one new operator: [`$dateToString`](https://docs.mongodb.com/manual/reference/operator/aggregation/dateToString/#exp._S_dateToString).
- [x] Use it to implement `TimeOfDay` in expr instead of Js (avoiding mapreduce).
- [ ] Possibly add a new date formatting function (possibly following the [PostgreSQL syntax](https://www.postgresql.org/docs/9.2/static/functions-formatting.html)).

Note: selectors now support [`$eq`](https://docs.mongodb.com/manual/reference/operator/query/eq/#op._S_eq), but the benefit of that for generated worklfows is probably not worth the effort to track types in `Selector`.

### PR DESCRIPTION
**Pull request by mossprescott**
Closes #1310.

The upshot here is just that `time_of_day` can now be evaluated in the aggregation pipeline. More interestingly, this lays the groundwork for the 3.2 ops to be generated as well.

### CODE REVIEW

File: foundation/src/main/scala/quasar/fp/tree/package.scala [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import matryoshka._
+import scalaz._
+
+/** "Generic" types for building partially-constructed trees in some
+  * "functorized" type. */
+// TODO: submit to matryoshka?
+package object tree {
+  /** A tree structure with one kind of hole. See [[UnaryOps.eval]]. */
+  type Unary[F[_]] = Free[F, UnaryArg]
+  object Unary {
```
Comment (mossprescott) [inline foundation/src/main/scala/quasar/fp/tree/package.scala]: @sellout: The idea is to replace `JsFn` with `Unary[JsCore]`, for example. Interesting?

**Comment by mossprescott** [comment]
Restarted the build ("server failed to start").

**Comment by wemrysi** [comment]
LGTM, I'll merge when green.

**Comment by jdegoes** [comment]
@mossprescott Remember to add at least one integration test for each new operator so we have some end-to-end coverage that benefits other connectors as well.

**Comment by mossprescott** [comment]
@jdegoes: In this case, we're just using the operator to implement an existing function in the pipeline instead of in JS, so one test now runs on the read-only Mongo configuration.

The more interesting operators in 3.2 will come with new tests.

**Pull request merged**